### PR TITLE
Always trigger testing for PRs from trusted users

### PR DESCRIPTION
**Pull request by fejta**
/assign @BenTheElder @krzyzacy @cblecker 

fixes https://github.com/kubernetes/test-infra/issues/8628

### CODE REVIEW

File: prow/plugins/trigger/ic.go [hunk 1]
```diff
@@ -109,7 +109,7 @@ func handleIC(c client, trigger *plugins.Trigger, ic github.IssueCommentEvent) e
 		if err != nil {
 			return fmt.Errorf("error listing issue comments: %v", err)
 		}
-		trusted, err := trustedPullRequest(c.GitHubClient, trigger, commentAuthor, org, repo, comments)
+		trusted, err := trustedPullRequest(c.GitHubClient, trigger, ic.Issue.User.Login, org, repo, comments)
```
Comment (BenTheElder) [inline prow/plugins/trigger/ic.go]: Side note for future consideration:
 Should we be do this for _all_ kinds of testing? `/retest` is limited to tests that failed and were already requested, but with `/test foo` you could spam requesting extra testing without even being a member..? /shrug

Comment (fejta) [reply prow/plugins/trigger/ic.go]: hypothetical problem thus far

Comment (BenTheElder) [reply prow/plugins/trigger/ic.go]: I mean so is malicious PR contents as far as I know. What's the use case for an untrusted commenter to trigger arbitrary tests?

Comment (fejta) [reply prow/plugins/trigger/ic.go]: It is about simpler code, not enabling use cases

Comment (fejta) [reply prow/plugins/trigger/ic.go]: The fact that something is technically more correct does not always make it better, if it requires higher implementation and maintenance costs.

I've seen a couple PRs that looked sketchy. And the risk/cost is a lot higher in the malicious PR scenario. Someone spamming `/test foo` doesn't really do anything aside from delay merges.

**Pull request merged**